File: ODE/1stODE.py
import timeit
import numpy as np
from scipy.integrate import odeint
import matplotlib.pyplot as plt

def dydt(u,x,k):
    return k * np.exp(-x)

y0 = 0
t = np.linspace(0,10,1000)

def solve_odeint():
    k = 0.1
    return odeint(dydt, y0, t, args=(k,))

sample=10000
execution_time = timeit.timeit(solve_odeint, number=sample)
print("Degree 1 Average Execution time:", execution_time/sample)

y = solve_odeint()
plt.plot(t, y, label='y')
plt.xlabel('t')
plt.ylabel('y')
plt.legend()
plt.show()

File: ODE/1stODE.jl
using DifferentialEquations
using BenchmarkTools
using Plots

function dydt(u, p, t)
    p * exp(-t)
end

t_span = (0.0, 10.0)
p = 0.1
u0 = 0
localprob = ODEProblem(dydt, u0, t_span, p)

num_sol = @benchmark solve($localprob)

println("Minimum time: $(minimum(num_sol.times) / 1e9) seconds")
println("Maximum time: $(maximum(num_sol.times) / 1e9) seconds")
println("Average time: $(mean(num_sol.times) / 1e9) seconds")

num_sol = solve(localprob)
plot(num_sol, xlabel="t", ylabel="y", label=["y"])
